test/sasdataloader/test/utest_cansas.py

```diff
@@ -250,23 +247,245 @@ def get_processing_instructions(self, xml_reader_object):
+        self.assertEqual(self.data.detector[0].distance, 4150)
+
+        self.assertEqual(self.data.detector[0].orientation_unit, "degree")
+        self.assertAlmostEqual(self.data.detector[0].orientation.x, 1.0, 6)
+        self.assertEqual(self.data.detector[0].orientation.y, 0.0)
+        self.assertEqual(self.data.detector[0].orientation.z, 0.0)
+
+        self.assertEqual(self.data.detector[0].offset_unit, "m")
+        self.assertEqual(self.data.detector[0].offset.x, .001)
+        self.assertEqual(self.data.detector[0].offset.y, .002)
+        self.assertEqual(self.data.detector[0].offset.z, None)
+
+        self.assertEqual(self.data.detector[0].beam_center_unit, "mm")
+        self.assertEqual(self.data.detector[0].beam_center.x, 322.64)
+        self.assertEqual(self.data.detector[0].beam_center.y, 327.68)
+        self.assertEqual(self.data.detector[0].beam_center.z, None)
+
+        self.assertEqual(self.data.detector[0].pixel_size_unit, "mm")
+        self.assertEqual(self.data.detector[0].pixel_size.x, 5)
+        self.assertEqual(self.data.detector[0].pixel_size.y, 5)
+        self.assertEqual(self.data.detector[0].pixel_size.z, None)
+
+        # Process
+        _found_term1 = False
+        _found_term2 = False
+        for item in self.data.process:
+            self.assertTrue(item.name in ['NCNR-IGOR', 'spol'])
+            self.assertTrue(item.date in ['04-Sep-2007 18:35:02',
+                                          '03-SEP-2006 11:42:47'])
+            for t in item.term:
+                if (t['name'] == "ABS:DSTAND"
+                    and t['unit'] == 'mm'
+                    and float(t['value']) == 1.0):
+                    _found_term2 = True
+                elif (t['name'] == "radialstep"
+                      and t['unit'] == 'mm'
+                      and float(t['value']) == 10.0):
+                    _found_term1 = True
+
+        if not _found_term1 or not _found_term2:
+            raise RuntimeError("Could not find all process terms %s %s"
+                               % (_found_term1, _found_term2))
+
+    def test_writer(self):
+        filename = "write_test.xml"
+        data_list = self.loader.load(self.cansas1d)
+        data = data_list[0]
+        self.cansas_reader.write(self.write_filename, data)
+        self.data_list = self.loader.load(self.write_filename)
+        self.data = self.data_list[0]
+        self.assertEqual(len(self.data_list), 1)
+        self.assertEqual(len(self.data_list), len(self.data_list))
+        self.assertEqual(self.data.filename, filename)
+        self._checkdata()
+        if os.path.isfile(self.write_filename):
+            os.remove(self.write_filename)
+
+    def test_units(self):
+        """
+            Check units.
+            Note that not all units are available.
+        """
+        filename = "cansas1d_units.xml"
+        self.data_list = self.cansas_reader.read(find(filename))
+        self.data = self.data_list[0]
```

Comment:
I added a load to this and all following tests that were part of the setUp() of the old class.